io.ossia.score.yml

```diff
@@ -0,0 +1,333 @@
+id: io.ossia.score
+runtime: org.kde.Platform
+runtime-version: '6.7'
+sdk: org.kde.Sdk
+command: ossia-score
+finish-args:
+  # BLE support
+  - --allow=bluetooth
+
+  # Access user's data and media libraries, plug-ins etc
+  - --filesystem=host
+
+  # GPU rendering support
+  - --device=dri
+
+  # Raw input support
+  - --device=input
+
+  # Required for USB & MIDI devices support as well as JACK (/dev/shm) and raw GPIO I/O
+  - --device=all
+
+  # Removable drives support
+  # https://github.com/flatpak/flatpak/issues/2713#issuecomment-1958510347
+  - --filesystem=/media
+  - --filesystem=/run/media
+
+  # X11 is used by default
+  - --socket=x11
+
+  # Wayland still has rendering regressions compared to X11 as of Qt 6.7
+  - --socket=wayland
+
+  # Used by X11
+  - --share=ipc
+
+  # Network access
+  - --share=network
+
+  # Permit PipeWire and user music directory access
+  - --filesystem=xdg-run/pipewire-0
+  - --filesystem=xdg-music
+
+  # PulseAudio is needed for ALSA device access
+  - --socket=pulseaudio
+
+  # Talk to UPower and RTKit services on the session bus
+  - --system-talk-name=org.freedesktop.UPower
+  - --system-talk-name=org.freedesktop.RealtimeKit1
+
+  # Ignore ALSA configuration
+  - --env=ALSA_CONFIG_PATH=
+  - --env=VST_PATH=/app/extensions/Plugins/vst
+  - --env=VST3_PATH=/app/extensions/Plugins/vst3
+
+  # Required to allow screensaver/idle inhibition
+  - --talk-name=org.freedesktop.ScreenSaver
+
+  # For correct cursor scaling under Wayland
+  - --env=XCURSOR_PATH=/run/host/user-share/icons:/run/host/share/icons
+
+  # Plugins
+  - --env=LV2_PATH=$HOME/.lv2:/app/extensions/Plugins/lv2:/app/lib/lv2
+
+  # Config persistence
+  - --persist=ossia
+
+add-extensions:
+  # Linux audio plugins base extension
+  org.freedesktop.LinuxAudio.Plugins:
+     directory: extensions/Plugins
+     version: '23.08'
+     add-ld-path: lib
+     merge-dirs: vst;vst3;lv2
+     subdirectories: true
+     no-autodownload: true
+
+  # Full codec support for FFMPEG
+  org.freedesktop.Platform.openh264:
+    directory: lib/openh264,
+    add-ld-path: .,
+    version: "2.4.1"
+
+  org.freedesktop.Platform.ffmpeg-full:
+    directory: lib/ffmpeg,
+    add-ld-path: .
+    version: "23.08"
+
+cleanup:
+  - /include
+  - /share/gir-1.0
+  - /share/man
+  - '*.a'
+  - '*.la'
+  - /lib/cmake
+  - /lib/girepository-1.0
+  - /lib/pkgconfig
+  - /lib/*.la
+  - /lib/*.a
+  - /share/pkgconfig
+  - /share/aclocal
+  - /share/gtk-doc
+  - /share/doc
+  - /share/info
+  - /share/man
+  - /man
+modules:
+  # LV2 audio plugin support
+  - shared-modules/linux-audio/lv2.json
+  - shared-modules/linux-audio/lilv.json
+
+  # Audio I/O
+  - name: portaudio
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
+      - -DCMAKE_POSITION_INDEPENDENT_CODE=ON
+      - -DBUILD_SHARED_LIBS=OFF
```

Comment:
in general all should be built shared.

Reply:
what's the value of this when no other process will be able to load these .so and thus won't benefit from deduplication ? This just adds indirection and increases binary size as the unused code cannot even be removed by the linker.

Reply:
also technically I can build most of the individual classes in the software as their own shared libs - would that bring any value ?

Reply:
deduplication is done at file level. That mean that if you update the main binary but not the dependencies then only the main binary is downloaded as part of the update. If you link everything static then you have humongus binaries.

The value is that we build libraries shared. There are a few excpetions, this is isn't one.

Reply:
> If you link everything static then you have humongus binaries.

for reference the appimage of ossia, which links against Qt, ffmpeg, libllvm, libclang, and all the libraries mentioned here is 101 megabytes and not particularly optimized - in contrast, with flatpak's shared libs approach there's 1.5G to install to get to not even the same feature level, that's a 15x difference in size.. so that's going to be a tough pill for my users which I'll have a hard time arguing for :) but fair, I changed the build to use everything shared.